### ISSUE TITLE
argument(s) mismatch for an intrinsic 'any()'. Wrong arg type

### ISSUE THREAD

**Issue by clementval** [omni-compiler/omni-compiler#302]
```fortran
MODULE mod1

  TYPE type_x
    INTEGER :: type
  END TYPE type_x

  TYPE type_y
    TYPE(type_x), ALLOCATABLE, DIMENSION(:) :: tx
  END TYPE type_y

  CONTAINS

  SUBROUTINE sub1()
    TYPE(type_y) :: var1
    INTEGER, PARAMETER :: TYPE1 = 1

    IF (ANY(var1%tx(1:10)%type == TYPE1) ) THEN 
    END IF
  END SUBROUTINE sub1
END MODULE mod1
```

In the example code above, the front-end fails with the following error message: 
```bash
line XX: argument(s) mismatch for an intrinsic 'any()'.
```

After a first investigation looks like the type of the arguments passed to `any()` is a `TYPE_LOGICAL` and `any()` accepts only LOGICAL ARRAY.

**Comment by clementval** [comment]
@h-murai @shingo-s the problem seems to be located in `compile_expression` the member ref is not detected as an array when compiling logical expression.

**Comment by shingo-s** [comment]
@clementval It seems that `compile_member_ref` has a problem.  In the line `6282`, this funciton copies the shape of the derived-type object to the expression via `copy_dimensions()`:
```c
        tp = compile_dimensions(tp, shape);
```
But in the above `var1%tx(1:10)%type`, F_Front doesn't come into this line.
So modifing the branch condition in line `6256` throgh `6258` may fix the bug.

**Comment by clementval** [comment]
@shingo-s Thanks for the input. I'll have a look at this.

**Comment by clementval** [comment]
@shingo-s 

Changing the condition from 

```c
if (!IS_FUNCTION_TYPE(ID_TYPE(member_id)) && (
        TYPE_HAS_SUBOBJECT_PROPAGATE_ATTRS(stVTyp) ||
        TYPE_IS_COINDEXED(stVTyp))) 
```

to

```c
if (!IS_FUNCTION_TYPE(ID_TYPE(member_id)))) 
```

fix the bug but it's probably not what we want right?

**Comment by shingo-s** [comment]
@clementval 

Your fix looks good. So please commit it.

And I found another problem for the case if `member_id` is a procedure, so I made an another issue as #310.
